tests/test_session.py

```diff
@@ -597,3 +597,28 @@ def test_heartbeat(self, make_manager):
         await sm._heartbeat_task()
         assert s1.id not in sm
         assert s2.id in sm
+
+    async def test_emits_warning_on_del(self, make_manager, make_session):
+
+        _, sm = make_manager()
+        s1 = make_session('id1')
+        s2 = make_session('id2')
+
+        sm._add(s1)
+        sm._add(s2)
+
+        with pytest.warns(RuntimeWarning) as warning:
+            getattr(sm, '__del__')()
```

Comment:
I believe `del sm` should be enough. `getattr()` usage looks too cryptic for my eyes.

Reply:
Tried that, and `__del__` was not triggered. Infact, in python `__del__` is not guaranteed to be called immediately. But with getattr we call the function without the GC cycle.